controllers/cloud.redhat.com/status.go

```diff
@@ -2,49 +2,183 @@ package controllers
 
 import (
 	"context"
+	"fmt"
 
 	"cloud.redhat.com/clowder/v2/controllers/cloud.redhat.com/object"
-	"cloud.redhat.com/clowder/v2/controllers/cloud.redhat.com/utils"
+	strimzi "github.com/RedHatInsights/strimzi-client-go/apis/kafka.strimzi.io/v1beta1"
 	apps "k8s.io/api/apps/v1"
-	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func filterOwnedDeployments(deploymentList *apps.DeploymentList, uid types.UID) {
-	depList := []apps.Deployment{}
-	for _, deployment := range deploymentList.Items {
-		for _, owner := range deployment.ObjectMeta.OwnerReferences {
-			if owner.UID == uid {
-				depList = append(depList, deployment)
+func deploymentStatusChecker(deployment apps.Deployment) bool {
```

Comment:
Using this now to parse the status conditions in a Deployment instead of utils.DeploymentStatusChecker

Reply:
We're now using the status conditions that we discussed in our call the other day, where we check for a condition of status: "true" and type: "Available" -- I thought we concluded that we could start doing that rather than checking the readyReplicas vs availableReplicas and whatnot that the older function was doing. Just simplifies the function (I think?)

Reply:
Aha, that's what I get for not reading your new function. I just assumed it was a reduced version of the other one.  Sounds good to me!